### DISCUSSION TITLE
Quire preview errors

### DISCUSSION THREAD

**Discussion by stephaniepaz03** [thegetty/quire#283] [Get Help] [answered]
Hello - Getting the  following errors and am unable to use the preview command. Any thoughts? I am running macOS Monterey.
[Terminal Saved Output.pdf](https://github.com/thegetty/quire/files/8524486/Terminal.Saved.Output.pdf)

**Comment by daniel-keller** [comment]
Hi @stephaniepaz03, I noticed in your output pdf shows you're using Node.js version `17.9.0`. I've had trouble running quire with newer versions of nodejs. I believe quire requires version `v14.18.1` which is a long term release supported until 2023. My suggestion would be to install `nvm` which is a handy utility for installing and switching between different versions of nodejs.

**Reply by daniel-keller** [reply]
Your output error also looks similar to the one reported in issue #164. It could be that you missed the step of installing dependencies by running `quire install` before `quire preview`. https://github.com/thegetty/quire/discussions/164#discussioncomment-1233931

**Reply by stephaniepaz03** [reply]
@daniel-keller thanks for the tip on Node.js  I did try the quire install before preview as mentioned in issue #164 and it did not work for me.

**Comment by Erin-Cecele** [comment]
Hi @stephaniepaz03 I believe the issue may be that the Monterey update removed Python 2 and so it needs to be reinstalled. 

Follow this link to install Python 2: https://www.python.org/downloads/release/python-2718/. Click the "macOS 64-bit installer" and follow the prompts. Then navigate to `themes/default` folder in your new project, delete the `node_modules` folder, then run `quire install` followed by `quire preview.`

Stephanie as @Daniel-keller suggests you should also make sure you are on the correct version of node which can be found here: https://nodejs.org/download/release/v14.18.1/node-v14.18.1.pkg. 

Please let me know if this resolves the issue and I will update the Quire documentation accordingly.

**Reply by stephaniepaz03** [reply]
Thanks! The combination of of the python 2 install and nodejs did enable me to preview. But, if you see attached, it doesn't look as it maybe should on the localhost. When I select the PDF version under 'other formats' it comes out normal, full color etc. I've also included the errors im getting 

<img width="809" alt="Screen Shot 2022-04-21 at 11 03 47 AM" src="https://user-images.githubusercontent.com/99911731/164497623-59e7c344-ee86-47bf-8461-7ae46c779877.png">

Last login: Thu Apr 21 11:21:41 on ttys000
stephaniepaz@Stephanies-MBP ~ % cd jaynecortez
stephaniepaz@Stephanies-MBP jaynecortez % sudo chown -R $(whoami) $(npm config get prefix)/{lib/node_modules,bin,share}
Password:
stephaniepaz@Stephanies-MBP jaynecortez % quire install
⠋ Installing theme dependencies...⸨░░░░░░░░░░░░░░░░░░⸩ ⠦ rollbackFailedOptional:
> fsevents@1.2.11 install /Users/stephaniepaz/jaynecortez/themes/default/node_modules/fsevents
> node-gyp rebuild

No receipt for 'com.apple.pkg.CLTools_Executables' found at '/'.

No receipt for 'com.apple.pkg.DeveloperToolsCLILeo' found at '/'.

No receipt for 'com.apple.pkg.DeveloperToolsCLI' found at '/'.

gyp: No Xcode or CLT version detected!
gyp ERR! configure error 
gyp ERR! stack Error: `gyp` failed with exit code: 1
gyp ERR! stack     at ChildProcess.onCpExit (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/configure.js:351:16)
gyp ERR! stack     at ChildProcess.emit (events.js:400:28)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:282:12)
gyp ERR! System Darwin 21.4.0
gyp ERR! command "/usr/local/bin/node" "/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /Users/stephaniepaz/jaynecortez/themes/default/node_modules/fsevents
gyp ERR! node -v v14.18.1
gyp ERR! node-gyp -v v5.1.0
gyp ERR! not ok 

> node-sass@4.13.0 install /Users/stephaniepaz/jaynecortez/themes/default/node_modules/node-sass
> node scripts/install.js

Downloading binary from https://github.com/sass/node-sass/releases/download/v4.13.0/darwin-x64-83_binding.node
Cannot download "https://github.com/sass/node-sass/releases/download/v4.13.0/darwin-x64-83_binding.node": 

HTTP error 404 Not Found

Hint: If github.com is not accessible in your location
      try setting a proxy via HTTP_PROXY, e.g. 

      export HTTP_PROXY=http://example.com:1234

or configure npm proxy via

      npm config set proxy http://example.com:8080

> node-sass@4.13.0 postinstall /Users/stephaniepaz/jaynecortez/themes/default/node_modules/node-sass
> node scripts/build.js

Building: /usr/local/bin/node /Users/stephaniepaz/jaynecortez/themes/default/node_modules/node-gyp/bin/node-gyp.js rebuild --verbose --libsass_ext= --libsass_cflags= --libsass_ldflags= --libsass_library=
gyp info it worked if it ends with ok
gyp verb cli [
gyp verb cli   '/usr/local/bin/node',
gyp verb cli   '/Users/stephaniepaz/jaynecortez/themes/default/node_modules/node-gyp/bin/node-gyp.js',
gyp verb cli   'rebuild',
gyp verb cli   '--verbose',
gyp verb cli   '--libsass_ext=',
gyp verb cli   '--libsass_cflags=',
gyp verb cli   '--libsass_ldflags=',
gyp verb cli   '--libsass_library='
gyp verb cli ]
gyp info using node-gyp@3.8.0
gyp info using node@14.18.1 | darwin | x64
gyp verb command rebuild []
gyp verb command clean []
gyp verb clean removing "build" directory
gyp verb command configure []
gyp verb check python checking for Python executable "python2" in the PATH
gyp verb `which` succeeded python2 /Library/Frameworks/Python.framework/Versions/2.7/bin/python2
gyp verb check python version `/Library/Frameworks/Python.framework/Versions/2.7/bin/python2 -c "import sys; print "2.7.18
gyp verb check python version .%s.%s" % sys.version_info[:3];"` returned: %j
gyp verb get node dir no --target version specified, falling back to host node version: 14.18.1
gyp verb command install [ '14.18.1' ]
gyp verb install input version string "14.18.1"
gyp verb install installing version: 14.18.1
gyp verb install --ensure was passed, so won't reinstall if already installed
gyp verb install version is already installed, need to check "installVersion"
gyp verb got "installVersion" 9
gyp verb needs "installVersion" 9
gyp verb install version is good
gyp verb get node dir target node version installed: 14.18.1
gyp verb build dir attempting to create "build" dir: /Users/stephaniepaz/jaynecortez/themes/default/node_modules/node-sass/build
gyp verb build dir "build" dir needed to be created? /Users/stephaniepaz/jaynecortez/themes/default/node_modules/node-sass/build
gyp verb build/config.gypi creating config file
gyp verb build/config.gypi writing out config file: /Users/stephaniepaz/jaynecortez/themes/default/node_modules/node-sass/build/config.gypi
gyp verb config.gypi checking for gypi file: /Users/stephaniepaz/jaynecortez/themes/default/node_modules/node-sass/config.gypi
gyp verb common.gypi checking for gypi file: /Users/stephaniepaz/jaynecortez/themes/default/node_modules/node-sass/common.gypi
gyp verb gyp gyp format was not specified; forcing "make"
gyp info spawn /Library/Frameworks/Python.framework/Versions/2.7/bin/python2
gyp info spawn args [
gyp info spawn args   '/Users/stephaniepaz/jaynecortez/themes/default/node_modules/node-gyp/gyp/gyp_main.py',
gyp info spawn args   'binding.gyp',
gyp info spawn args   '-f',
gyp info spawn args   'make',
gyp info spawn args   '-I',
gyp info spawn args   '/Users/stephaniepaz/jaynecortez/themes/default/node_modules/node-sass/build/config.gypi',
gyp info spawn args   '-I',
gyp info spawn args   '/Users/stephaniepaz/jaynecortez/themes/default/node_modules/node-gyp/addon.gypi',
gyp info spawn args   '-I',
gyp info spawn args   '/Users/stephaniepaz/.node-gyp/14.18.1/include/node/common.gypi',
gyp info spawn args   '-Dlibrary=shared_library',
gyp info spawn args   '-Dvisibility=default',
gyp info spawn args   '-Dnode_root_dir=/Users/stephaniepaz/.node-gyp/14.18.1',
gyp info spawn args   '-Dnode_gyp_dir=/Users/stephaniepaz/jaynecortez/themes/default/node_modules/node-gyp',
gyp info spawn args   '-Dnode_lib_file=/Users/stephaniepaz/.node-gyp/14.18.1/<(target_arch)/node.lib',
gyp info spawn args   '-Dmodule_root_dir=/Users/stephaniepaz/jaynecortez/themes/default/node_modules/node-sass',
gyp info spawn args   '-Dnode_engine=v8',
gyp info spawn args   '--depth=.',
gyp info spawn args   '--no-parallel',
gyp info spawn args   '--generator-output',
gyp info spawn args   'build',
gyp info spawn args   '-Goutput_dir=.'
gyp info spawn args ]
No receipt for 'com.apple.pkg.CLTools_Executables' found at '/'.

No receipt for 'com.apple.pkg.DeveloperToolsCLILeo' found at '/'.

No receipt for 'com.apple.pkg.DeveloperToolsCLI' found at '/'.

gyp: No Xcode or CLT version detected!
gyp ERR! configure error 
gyp ERR! stack Error: `gyp` failed with exit code: 1
gyp ERR! stack     at ChildProcess.onCpExit (/Users/stephaniepaz/jaynecortez/themes/default/node_modules/node-gyp/lib/configure.js:345:16)
gyp ERR! stack     at ChildProcess.emit (events.js:400:28)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:282:12)
gyp ERR! System Darwin 21.4.0
gyp ERR! command "/usr/local/bin/node" "/Users/stephaniepaz/jaynecortez/themes/default/node_modules/node-gyp/bin/node-gyp.js" "rebuild" "--verbose" "--libsass_ext=" "--libsass_cflags=" "--libsass_ldflags=" "--libsass_library="
gyp ERR! cwd /Users/stephaniepaz/jaynecortez/themes/default/node_modules/node-sass
gyp ERR! node -v v14.18.1
gyp ERR! node-gyp -v v3.8.0
gyp ERR! not ok 
Build failed with error code: 1
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.2.11 (node_modules/fsevents):
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.2.11 install: `node-gyp rebuild`
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: Exit status 1

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! node-sass@4.13.0 postinstall: `node scripts/build.js`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the node-sass@4.13.0 postinstall script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/stephaniepaz/.npm/_logs/2022-04-21T15_25_15_500Z-debug.log
✔ Theme dependencies successfully installed.
✔ Run quire preview to view changes locally.
stephaniepaz@Stephanies-MBP jaynecortez % quire preview
Launching preview server
ℹ Navigate to http://localhost:1313 to see your changes.
ℹ Press Control+C or type "quire stop" to stop the preview.
10% building 0/0 modules 0 active
webpack is watching the files…

10% building 0/1 modules 1 active ...ez/themes/default/source/js/application.jsBrowserslist: caniuse-lite is outdated. Please run next command `npm update`
^CShutting down Quire
stephaniepaz@Stephanies-MBP jaynecortez % ^[[200~npm install --global @thegetty/quire-cli~
zsh: bad pattern: ^[[200~npm
stephaniepaz@Stephanies-MBP jaynecortez % npm install --global @thegetty/quire-cli
npm WARN deprecated axios@0.19.2: Critical security vulnerability fixed in v0.21.1. For more information, see https://github.com/axios/axios/pull/3410
npm WARN deprecated source-map-resolve@0.5.3: See https://github.com/lydell/source-map-resolve#deprecated
npm WARN deprecated source-map-url@0.4.1: See https://github.com/lydell/source-map-url#deprecated
npm WARN deprecated resolve-url@0.2.1: https://github.com/lydell/resolve-url#deprecated
npm WARN deprecated urix@0.1.0: Please see https://github.com/lydell/urix#deprecated
npm WARN deprecated querystring@0.2.0: The querystring API is considered Legacy. new code should use the URLSearchParams API instead.
npm WARN deprecated chokidar@2.1.8: Chokidar 2 does not receive security updates since 2019. Upgrade to chokidar 3 with 15x fewer dependencies
npm WARN deprecated fsevents@1.2.13: fsevents 1 will break on node v14+ and could be using insecure binaries. Upgrade to fsevents 2.
/usr/local/bin/quire -> /usr/local/lib/node_modules/@thegetty/quire-cli/bin/index.js
+ @thegetty/quire-cli@0.20.2
updated 1 package in 20.787s
stephaniepaz@Stephanies-MBP jaynecortez % quire preview
Launching preview server
ℹ Navigate to http://localhost:1313 to see your changes.
ℹ Press Control+C or type "quire stop" to stop the preview.
10% building 0/0 modules 0 active
webpack is watching the files…

10% building 0/1 modules 1 active ...ez/themes/default/source/js/application.jsBrowserslist: caniuse-lite is outdated. Please run next command `npm update`
^CShutting down Quire
stephaniepaz@Stephanies-MBP jaynecortez %

**Reply by daniel-keller** [reply]
From the error message it looks like `quire install` failed to install all the dependences. I see the line `No Xcode or CLT version detected!`. It could be that apple's XCode isn't installed or didn't install properly. If you've already run `xcode-select --install` try reseting xcode with `xcode-select --reset`. If that doesn't correct the issue you could try updating the xcode path as described here https://stackoverflow.com/a/62930662. As a last resort you can try uninstalling and reinstalling xcode (it can take awhile to install).

**Reply by stephaniepaz03** [reply]
@daniel-keller thanks!! i ran the `xcode-select --install` and followed Erin's steps below after and got preview to work!

**Comment by Erin-Cecele** [comment, marked as answer]
Hi @stephaniepaz03 - @daniel-keller is right, it looks like there is an issue with XCode. Let us know if any of the suggestions Daniel made end up working for you. (Thank you for all the assistance @daniel-keller!!) 

Once you have done that please try following these steps again: Navigate to `themes/default` folder in your new project, delete the `node_modules` folder, then run `quire install` followed by `quire preview`. You should **not** need to reinstall Quire itself or run the `sudo chown` command.

**Reply by stephaniepaz03** [reply]
Thanks again, Erin! I followed your steps after @daniel-keller 's xcode suggestions and got preview working!
